web-ui/src/components/feedback_submit_form/FeedbackSubmitForm.jsx

```diff
@@ -76,65 +75,177 @@ const ColorButton = withStyles({
 const propTypes = {
   requesteeName: PropTypes.string.isRequired,
   requestId: PropTypes.string.isRequired,
+  request: PropTypes.any.isRequired,
 }
 
-const FeedbackSubmitForm = (props) => {
+const FeedbackSubmitForm = ({ requesteeName, requestId, request }) => {
+  const { state, dispatch } = useContext(AppContext);
+  const csrf = selectCsrfToken(state);
   const classes = useStyles();
-  const handleClick = () => history.push(`/feedback/submit/confirmation/?request=${props.requestId}`);
   const [isReviewing, setIsReviewing] = useState(false);
   const history = useHistory();
+  const [questionAnswerPairs, setQuestionAnswerPairs] = useState([])
+  let currentlyBeingEdited = -1
+
+
+  const updateAnswer = debounce(async () => {
+    if (csrf) {
+      const res = await updateSingleAnswer(questionAnswerPairs[currentlyBeingEdited].answer, csrf)
+      return res;
+    }
+  }, 2000)
+
+
+  async function updateRequestSubmit() {
+    request.status = "submitted"
+    request.submitDate = dateUtils.format(new Date(), "yyyy-MM-dd")
+    const res = await updateFeedbackRequest(request, csrf);
+    return res;
+
+  }
+  async function updateAllAnswersSubmit(){
+    let answers= [];
+    for (let i = 0; i < questionAnswerPairs.length; ++i) {
+      answers.push(questionAnswerPairs[i].answer)
+    }
+    const res = await updateAllAnswers(answers, csrf)
+    return res;
+  }
+
+  const onSubmitHandler =() => {
+    updateAllAnswersSubmit().then((res) => {
+      for (let i = 0; i < res.length; ++i ) {
+        if (res[i].error) {
+          dispatch({
+            type: UPDATE_TOAST,
+            payload: {
+              severity: "error",
+              toast: res[i].error,
+            },
+          });
+          return false;
+        }
+      }
+      return true;
+    }).then((resTwo) => {
+      if (resTwo === false) {
+        return;
+      }
```

Comment:
What kind of information is `resTwo`? This strict check against `false` will only work if it's a boolean

Reply:
It is a boolean that determines whether or not all answers have successfully gone to database.